sql/maintenance_utils.sql

```diff
@@ -117,3 +117,43 @@ BEGIN
     END CASE;
 END
 $$ LANGUAGE plpgsql;
+
+-- A version of makeaclitem that accepts a comma-separated list of
+-- privileges rather than just a single privilege. This is copied from
+-- PG16, but since we need to support earlier versions, we provide it
+-- with the extension.
+--
+-- This is intended for internal usage and interface might change.
+CREATE FUNCTION _timescaledb_functions.makeaclitem(regrole, regrole, text, bool)
+RETURNS AclItem AS '@MODULE_PATHNAME@', 'ts_makeaclitem'
+LANGUAGE C STABLE PARALLEL SAFE;
+
+-- Repair relation ACL by removing roles that do not exist in pg_authid.
+CREATE PROCEDURE _timescaledb_functions.repair_relation_acls()
+LANGUAGE SQL AS $$
+  WITH
+    badrels AS (
+	SELECT oid::regclass
+	  FROM (SELECT oid, (aclexplode(relacl)).* FROM pg_class) AS rels
+	 WHERE rels.grantee != 0
+	   AND rels.grantee NOT IN (SELECT oid FROM pg_authid)
+    ),
+    pickacls AS (
+      SELECT oid::regclass,
+	     _timescaledb_functions.makeaclitem(
+	         b.grantee,
+		 b.grantor,
+		 string_agg(b.privilege_type, ','),
+		 b.is_grantable
+	     ) AS acl
+	FROM (SELECT oid, (aclexplode(relacl)).* AS a FROM pg_class) AS b
+       WHERE b.grantee IN (SELECT oid FROM pg_authid)
+       GROUP BY oid, b.grantee, b.grantor, b.is_grantable
```

Comment:
```suggestion
    rels AS (
        SELECT oid, (aclexplode(relacl)).* FROM pg_class
    ),
    badrels AS (
	SELECT rels.oid::regclass
	  FROM rels
	 WHERE rels.grantee != 0
	   AND NOT EXISTS (SELECT FROM pg_authid a WHERE a.oid = rels.grantee)
    ),
    pickacls AS (
      SELECT DISTINCT
             oid::regclass,
	     _timescaledb_functions.makeaclitem(
	         rels.grantee,
		 rels.grantor,
		 string_agg(rels.privilege_type, ','),
		 rels.is_grantable
	     ) AS acl
	FROM rels
	JOIN pg_authid a ON a.oid = rels.grantee
```
Nit. Just small refactoring but it is up to you!

Reply:
Sorry, but this one do not work. It is important to group by grantee and grantor since there can be different ones used in the catalog.  Consider, for example, the case where different privileges are granted by different grantors to different grantee. It is not necessary to group by `is_grantable`, but it will not make a difference in practice.